pkg/api/obligationTypes.go

```diff
@@ -0,0 +1,280 @@
+	if err := validate.Struct(&obType); err != nil {
+		er := models.LicenseError{
+			Status:    http.StatusBadRequest,
+			Message:   "can not create obligation type with these field values",
+			Error:     fmt.Sprintf("field '%s' failed validation: %s\n", err.(validator.ValidationErrors)[0].Field(), err.(validator.ValidationErrors)[0].Tag()),
+			Path:      c.Request.URL.Path,
+			Timestamp: time.Now().Format(time.RFC3339),
+		}
+		c.JSON(http.StatusBadRequest, er)
+		return
+	}
+
+	if err := db.DB.Transaction(func(tx *gorm.DB) error {
+		result := tx.Where(&models.ObligationType{Type: obType.Type}).FirstOrCreate(&obType)
+		if result.Error != nil {
+			er := models.LicenseError{
+				Status:    http.StatusInternalServerError,
+				Message:   "something went wrong while creating new obligation type",
+				Error:     result.Error.Error(),
+				Path:      c.Request.URL.Path,
+				Timestamp: time.Now().Format(time.RFC3339),
+			}
+			c.JSON(http.StatusInternalServerError, er)
+			return result.Error
+		}
+		if result.RowsAffected == 0 {
+			if *obType.Active {
+				er := models.LicenseError{
+					Status:    http.StatusConflict,
+					Message:   "obligation type already exists",
+					Error:     "obligation type already exists",
+					Path:      c.Request.URL.Path,
+					Timestamp: time.Now().Format(time.RFC3339),
+				}
+				c.JSON(http.StatusConflict, er)
+				return errors.New("obligation type already exists")
+			}
+			if err := toggleObligationTypeActiveStatus(c, tx, &obType); err != nil {
+				er := models.LicenseError{
+					Status:    http.StatusConflict,
+					Message:   "obligation type already exists, something went wrong while reactvating it",
+					Error:     err.Error(),
+					Path:      c.Request.URL.Path,
+					Timestamp: time.Now().Format(time.RFC3339),
+				}
+				c.JSON(http.StatusConflict, er)
+				return err
+			}
+		}
+		return nil
+	}); err != nil {
+		return
+	}
+
+	res := models.ObligationTypeResponse{
+		Status: http.StatusCreated,
+		Data:   []models.ObligationType{obType},
+		Meta: &models.PaginationMeta{
+			ResourceCount: 1,
+		},
+	}
+
+	c.JSON(http.StatusCreated, res)
+}
+
+// DeleteObligationType marks an existing obligation type record as inactive
+//
+//	@Summary		Deactivate obligation type
+//	@Description	Deactivate an obligation type
+//	@Id				DeleteObligationType
+//	@Tags			Obligations
+//	@Accept			json
+//	@Produce		json
+//	@Param			type	path	string	true	"Obligation Type"
+//	@Success		200
+//	@Failure		400	{object}	models.LicenseError	"cannot delete obligation type 'RISK' as it's still referenced by some obligations"
+//	@Failure		404	{object}	models.LicenseError	"obligation type 'RISK' not found"
+//	@Failure		500	{object}	models.LicenseError	"something went wrong while deleting obligation type"
+//	@Security		ApiKeyAuth
+//	@Router			/obligations/types/{type} [delete]
+func DeleteObligationType(c *gin.Context) {
+	var obType models.ObligationType
+	obTypeParam := c.Param("type")
+	active := true
```

Comment:
Let us not restrict user to call this endpoint only on active types. This will confuse them seeing a 404

Even if type is already deactivated, please proceed as normal if there are no side effects (maybe check if already deactivated, return 200).